capella_ros_tools/exporter.py

```diff
@@ -2,83 +2,625 @@
-            type_def = data_model.TypeDef(name=prop_obj.type.name, card=card)
-            prop_def = data_model.FieldDef(
-                type=type_def,
-                name=prop_obj.name,
-                description=prop_obj.description or "",
+def int_bytes(length: int) -> int:
+    """Return ROS byte length for integers."""
+    for int_length in INT_LENGTHS:
+        if length <= int_length:
+            return int_length
+    raise ValueError(f"Invalid integer length {length}")
+
+
+def float_bytes(length: int) -> int:
+    """Return ROS byte length for floats."""
+    for float_length in FLOAT_LENGTHS:
+        if length <= float_length:
+            return float_length
+    raise ValueError(f"Invalid float length {length}")
+
+
+@dataclasses.dataclass
+class LiteralData:
+    """Data for class properties and enum values."""
+
+    type: str
+    name: str
+    value: str | None = None
+    docstr: list[str] = dataclasses.field(default_factory=list)
+
+
+@dataclasses.dataclass
+class ClassData:
+    """Data for a class."""
+
+    name: str
+    docstr: list[str] = dataclasses.field(default_factory=list)
+    literals: list[LiteralData] = dataclasses.field(default_factory=list)
+
+
+class MyHTMLParser(HTMLParser):
+    """An HTML parser to convert an HTML string to a list of plain strings."""
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.text_list: list[str] = []
+
+    def handle_data(self, data: str) -> None:
+        """Process data and fill the text list."""
+        if data.strip():  # Skipping empty strings
+            self.text_list.append(data.strip())
+
+
+class Exporter:
+    """The exporter class."""
+
+    packages: dict[str, str]  # mapping from ROS pkg name to capella pkg uuid
+    build_ins: dict[str, str]  # maps build in ROS pkgs to capella pkg uuids
+    package_uuids: list[str]  # list of all pkg uuids
+    custom_pkg: dict[
+        str, list[information.Class]
+    ]  # maps ROS pkg names to capella classes
+    custom_types: dict[str, str]  # maps custom capella types to ros types
+
+    def __init__(
+        self,
+        packages: dict[str, str],
+        build_ins: dict[str, str],
+        custom_pkg: dict[str, list[str]],
+        custom_types: dict[str, str],
+        model: capellambse.MelodyModel,
+        generate_cmake: bool,  # noqa: FBT001
+        pkg_postfix: str | None = None,
+    ) -> None:
+        self.generate_cmake = generate_cmake
+        self.model = model
+        self.packages = packages
+        self.build_ins = build_ins
+        self.package_uuids = list(packages.values()) + list(build_ins.values())
+        self.custom_types = custom_types
+        self.custom_pkg = {
+            pkg: [self.model.by_uuid(uuid) for uuid in uuids]
+            for pkg, uuids in custom_pkg.items()
+        }
+        self.jinja_env = jinja2.Environment(
+            loader=jinja2.FileSystemLoader(
+                pathlib.Path(__file__).parent / "export_templates"
```

Comment:
Technically we should be using the `PackageLoader` here.


```suggestion
            loader=jinja2.PackageLoader(
                __name__.rsplit(".", 1)[0], "export_templates"
```